modules/activemq-artemis-launch/added/drain.sh

```diff
@@ -134,5 +134,35 @@ if [ "${HTTP_CODE}" != "200" ]; then
   exit 1
 fi
 
+#restart the broker to check messages
```

Comment:
In normal circumstances the broker will stop but do we need to wait for stopped?

I am wondering if there is a possibility that the restart fails if the broker is still stopping

Reply:
The restart command actually do a stop and start. The stop will wait for the process to end. It's like this
```
stop() {
  if [ -f "${PID_FILE}" ] ; then
    pid=`cat "${PID_FILE}"`
    kill $@ ${pid} > /dev/null
  fi
  for i in 1 2 3 4 5 ; do
    status
    if [ $? -ne 0 ] ; then
      return 0
    fi
    sleep 1
  done
  echo "Could not stop process ${pid}"
  return 1
}
```

Reply:
there is a force-stop, https://github.com/apache/activemq-artemis/blob/cd1e9822f597b33474883c3c09397b0a7ef82f8c/artemis-cli/src/main/resources/org/apache/activemq/artemis/cli/commands/bin/artemis-service#L124

maybe we need to use stop, and if it fails use force-stop, then start. (as in, have have to do our own restart logic) or maybe fix the restart script.

Reply:
makes sense. I'll use force stop. Thanks!

Reply:
@gtully Updated the branch. Pls review.